### PR TITLE
IE9 fix for cross domain calls

### PR DESCRIPTION
**Pull request by ruchigoyal2005**
This pull request includes these fixes for IE9:
1) Fixed axios promise rejection on request onload for cors api calls.
2) Fixed cors api calls timing out prematurely.

### CODE REVIEW

**Comment by nettofarah** [comment]
I'm assuming that this works in all browsers (because travis is passing)
but have you manually checked this in other browsers too?

Also, can I suggest that you rewrite your commit message to make it a little more clear?
Here's a great guide on how to write good commit messages: http://chris.beams.io/posts/git-commit/

**Comment by ruchigoyal2005** [comment]
I tested this change manually in Chrome/firefox/IE9/IE10/IE11. I updated the commit message. thanks for the suggestion !!!

File: lib/adapters/xhr.js [hunk 1]
```diff
@@ -35,6 +35,10 @@ module.exports = function xhrAdapter(resolve, reject, config) {
   // Set the request timeout in MS
   request.timeout = config.timeout;
 
+  // For IE 9 CORS support.
+  request.onprogress = function() {};
```
Comment (mzabriskie) [inline lib/adapters/xhr.js]: What is the purpose of handling these events with noop functions?

**Comment by mzabriskie** [comment]
fixes #240

**Comment by mzabriskie** [comment]
How does this work with the reference to `request.responseText`? This seems really strange to get the response from the request.

**Comment by ruchigoyal2005** [comment]
Request is XDomainRequest and it has responseText property which retrieves the response body as a string.
https://msdn.microsoft.com/en-us/library/cc288060(v=vs.85).aspx

**Comment by mzabriskie** [comment]
It's only `XDomainRequest` if it exists, otherwise it's `XMLHttpRequest`. Will it still work in both scenarios?

**Comment by ruchigoyal2005** [comment]
@mzabriskie : For XMLHttpRequest we do get the response status. This scenario exists only for XDomainRequest. If you see this condition:

((response.status >= 200 && response.status < 300) ||
     (!('status' in request) && request.responseText) ?
      resolve :
      reject)(response);

In 1st condition response status getting checked. For XMLHttpReuqest this 1st condition gets executed. For XDomainRequest we don't get the status in the response, that's why we need to check the request.responseText.

**Comment by ruchigoyal2005** [comment]
Can you please merge this pull request if it looks ok ?..thanks !

**Comment by mzabriskie** [comment]
Thank you for your PR!

**Comment by nickuraltsev** [comment]
@ruchigoyal2005 @mzabriskie I'm still curious why this code rejects the promise in case when `request.responseText` is empty. For example, it's common to return an empty response body for successful PUT and DELETE requests.

**Pull request merged**